File: csharp/BruteForceIE.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace inference_algorithm
{
    public class BruteForceIE : AbstractIE
    {
        protected override void assertFact(object fact)
        {
            throw new NotImplementedException();
        }

        protected override void clearRuntimeFactStorage()
        {
            throw new NotImplementedException();
        }

        protected override List<object> getAllFacts()
        {
            throw new NotImplementedException();
        }

        protected override List<object> getAllFactsInRuntimeStorage()
        {
            throw new NotImplementedException();
        }

        protected override List<object> getAllRules()
        {
            throw new NotImplementedException();
        }

        protected override string getConditionExpression(object condition)
        {
            throw new NotImplementedException();
        }

        protected override object getConditionParameter(object condition)
        {
            throw new NotImplementedException();
        }

        protected override string getConsequenceExpression(object consequence)
        {
            throw new NotImplementedException();
        }

        protected override object getConsequenceParameter(object consequence)
        {
            throw new NotImplementedException();
        }

        protected override string getFactName(object fact)
        {
            throw new NotImplementedException();
        }

        protected override List<object> getFactProperties(object fact)
        {
            throw new NotImplementedException();
        }

        protected override string getFactPropertyName(object property)
        {
            throw new NotImplementedException();
        }

        protected override object getFactPropertyOnPosition(object fact, int position)
        {
            throw new NotImplementedException();
        }

        protected override string getFactPropertyType(object property)
        {
            throw new NotImplementedException();
        }

        protected override object getFactPropertyValue(object property)
        {
            throw new NotImplementedException();
        }

        protected override int getNrOfFactProperties(object fact)
        {
            throw new NotImplementedException();
        }

        protected override List<object> getRuleConditions(object rule)
        {
            throw new NotImplementedException();
        }

        protected override List<object> getRuleConsequences(object rule)
        {
            throw new NotImplementedException();
        }

        protected override RuleOperator getRuleOperator(object rule)
        {
            throw new NotImplementedException();
        }

        protected override bool isFactPropertyMandatory(object property)
        {
            throw new NotImplementedException();
        }

        protected override bool propertyRequiresInputFromUser(object property)
        {
            throw new NotImplementedException();
        }

        protected override void setConditionParameter(object condition, object parameter)
        {
            throw new NotImplementedException();
        }

        protected override void setFactPropertyValue(object property, object value)
        {
            throw new NotImplementedException();
        }
    }
}

File: javascript/bruteforceIE.js

; var bruteforceIE = (function() {
    var module = {};

	module.getAllRules = function(){
        // TODO
    }

	module.getRuleOperator = function(rule){
        // TODO
    }

	module.getRuleConditions = function(rule) {
        // TODO
    }

	module.getRuleConsequences = function(rule) {
        // TODO
    }

	module.getConditionExpression = function(condition) {
        // TODO
    }

	module.getConsequenceExpression = function(consequence) {
        // TODO
    }

    	/**
	 * The returned parameter can be a fact object, or a String.
	 * 
	 * @param consequence
	 * @return
	 */
	module.getConsequenceParameter = function(consequence) {
        // TODO
    }

	module.getConditionParameter = function(condition) {
        // TODO
    }

    /**
	 * Completely clean runtime storage.
	 */
	module.clearRuntimeFactStorage = function() {
        // TODO
    }

	module.getAllFactsInRuntimeStorage = function() {
        // TODO
    }

	/**
	 * Put new fact to runtime storage.
	 * 
	 * @param fact
	 */
	module.assertFact = function(fact) {
        // TODO
    }

	module.getFactProperties = function(fact) {
        // TODO
    }

	module.getFactName = function(fact) {
        // TODO
    }

	module.propertyRequiresInputFromUser = function(property) {
        // TODO
    }

	/**
	 * This method MUST return a valid fully qualified java class name!
	 * 
	 * @param property
	 * @return
	 */
	module.getFactPropertyType = function(property) {
        // TODO
    }

	module.setFactPropertyValue = function(property, value) {
        // TODO
    }

	module.getFactPropertyName = function(property) {
        // TODO
    }

	module.setConditionParameter = function(condition, parameter) {
        // TODO
    }

	module.getAllFacts = function() {
        // TODO
    }

	module.getNrOfFactProperties = function(fact) {
        // TODO
    }

	module.getFactPropertyOnPosition = function(fact, position) {
        // TODO
    }

	module.isFactPropertyMandatory = function(property) {
        // TODO
    }

	module.getFactPropertyValue = function(property) {
        // TODO
    }

    return module;
});